### PR TITLE
Added an example card integration

### PR DESCRIPTION
**Pull request by SayemW**

### CODE REVIEW

File: example-card-integration/data-connector/README.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Example Data Connector
+This integration uses a data connector to fetch and update data from the mock data source. 
+The data connector aslo defines how data will be represented when it gets added to Lucid documents.
+
+You can read more about developing data connectors [here](https://developer.lucid.co/extension-api/#connecting-to-external-data).
+
+## Folder / file structure
```
Comment (blackburn32) [inline example-card-integration/data-connector/README.md]: Nit: file -> File

File: example-card-integration/data-connector/README.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+```
+└── data-connector
+    ├── actions
+    ├── schema
+```
+
+`actions` contains the logic for all of the requests that are made to the data connector, in this case imports & patches (when data updates need to be sent to the external source)
```
Comment (blackburn32) [inline example-card-integration/data-connector/README.md]: Nit: add `.` to end

File: example-card-integration/data-connector/actions/importaction.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+export const importAction: (action: DataConnectorAsynchronousAction) => Promise<{success: boolean}> = async (
+    action,
+) => {
+    const primaryKeys = (action.data as any).taskIds as string[];
+    const itemsToAdd: ExampleTaskType[] = getTaskData(primaryKeys).filter((item) =>
+        primaryKeys.some((key) => JSON.stringify(item.id) === key),
+    );
+
+    action.client.update({
```
Comment (blackburn32) [inline example-card-integration/data-connector/actions/importaction.ts:15]: Nit: I think we should use `declareSchema` in here so people know they can use that (it also provides the `forItem` helper for primary keys)

File: example-card-integration/data-connector/actions/patchaction.ts [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+import {ExampleTaskType} from '../schema/examplecollectionschema';
+import {fullTaskDataToLucidFormat} from '../schema/exampletaskutil';
+
+export const patchAction: (action: DataConnectorPatchAction) => Promise<PatchChange[]> = async (action) => {
+    const extenalExampleDataClient = new ExternalExampleDataClient();
+    return await Promise.all(
+        action.patches.map(async (patch) => {
+            const change = patch.getChange();
+            await Promise.all([
+                ...Object.entries(patch.itemsAdded).map(async ([oldPrimaryKey, additions]) => {
```
Comment (blackburn32) [inline example-card-integration/data-connector/actions/patchaction.ts]: Nit: this huge promise.all statements gets pretty nested. Consider breaking down into helpers

File: example-card-integration/data-connector/package.json [hunk 5]
```diff
@@ -0,0 +1,30 @@
+{
+  "name": "data-connector",
+  "version": "0.0.1",
+  "description": "",
+  "keywords": [],
+  "author": "Lucid Software",
+  "license": "Apache-2.0",
+  "devDependencies": {
+    "@types/asana": "^0.18.10",
```
Comment (blackburn32) [inline example-card-integration/data-connector/package.json:9]: I'm guessing we don't need this

File: example-card-integration/data-connector/package.json [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    "@types/node": "^16.11.11",
+    "nodemon": "^2.0.19",
+    "ts-node": "^10.9.1",
+    "typescript": "^4.9.5"
+  },
+  "publishConfig": {
+    "access": "public"
+  },
+  "dependencies": {
+    "@types/asana": "^0.18.10",
```
Comment (blackburn32) [inline example-card-integration/data-connector/package.json]: Same here

File: example-card-integration/editorextensions/example-card-integration/src/examplecardintegrationtaskcreator.ts [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+        input: Map<string, SerializedFieldType>,
+    ): Promise<{collection: CollectionProxy; primaryKey: string}> {
+        let collection: CollectionProxy;
+        try {
+            // Check if the Tasks collection has already been created in the example-tasks-source data source,
+            // with a 1ms timeout. If the collection doesn't exist yet, an exception will be thrown.
+            collection = await this.client.awaitDataImport(DataConnecorName, DataSourceName, CollectionName, [], 1);
+        } catch {
+            // No collection exists yet. Ask the data connector to perform an import of an empty list of tasks,
+            // ust to get the collection created along with whatever other metadata you might need to set up there.
```
Comment (blackburn32) [inline example-card-integration/editorextensions/example-card-integration/src/examplecardintegrationtaskcreator.ts]: looks like the `j` on the beginning of this comment got cut

**Review by blackburn32 — APPROVED** [review]
Overall, great work, just had a couple nit picks

File: example-card-integration/data-connector/actions/importaction.ts (hunk 3 above)
Comment (SayemW) [reply example-card-integration/data-connector/actions/importaction.ts:15]: exampleCollectionSchema uses declareSchema

**Pull request merged**